backend/src/helper/index.js

```diff
@@ -11,6 +11,21 @@ export function findInObject(data, searchParameter, exclude){
     })
 }
 
+export function filterByRelation(data, searchParameter, relation){
+    return JSON.parse(JSON.stringify(data)).filter(function (parameter) {
+        return Object.keys(searchParameter).every(function (key) {
+            return relation(parameter[key], searchParameter[key]);
+        })
+    })
+}
+
+export function mergeObject(data, searchParameter1, searchParameter2) {
```

Comment:
well done, you can use Object.assign() to merge two objects or {...object1, ...Object2} which are native to javascript so you will not need to create your own method to handle that